streams/stream.definition.schema.json

```diff
@@ -0,0 +1,83 @@
+{
+  "$id": "https://streamofjava.github.io/stream-definition.schema.json",
+  "$schema": "https://json-schema.org/draft/2019-09/schema",
+  "description": "Information about the stream schedule of a Twitch channel",
+  "type": "object",
+  "properties": {
+    "twitch_handle": {
+      "type": "string"
+    },
+    "youtube_handle": {
+      "type": "string"
+    },
+    "twitter_handle": {
+      "type": "string"
+    },
+    "streamer_timezone": {
+      "type": "string"
+    },
+    "schedule": {
+      "type": "array",
+      "items": {
+        "type": "object",
+        "properties": {
+          "title": {
+            "type": "string"
+          },
+          "start_date_utc": {
+            "type": "string"
+          },
+          "duration_in_minutes": {
+            "type": "number"
+          },
+          "platform": {
+            "type": "string",
+            "enum": [
+              "twitch",
+              "youtube"
+            ]
+          },
+          "stream_link": {
+            "type": "string"
+          },
+          "recurring": {
+            "type": "object",
+            "properties": {
+              "is_recurring": {
+                "type": "boolean"
+              },
+              "series_end": {
+                "type": "string"
+              }
+            },
+            "required": [
+              "is_recurring",
+              "series_end"
+            ]
+          }
+        },
+        "required": [
+          "title",
+          "start_date_utc",
+          "duration_in_minutes",
+          "platform",
+          "stream_link"
+        ]
+      }
+    }
+  },
+  "anyOf": [
+    {
+      "required": [
+        "twitch_handle",
+        "schedule"
+      ]
+    },
+    {
+      "required": [
+        "youtube_handle",
+        "schedule"
+      ]
+    }
+  ]
```

Comment:
I don't know JSON schema, but I gathered only `twitch_handle` or `youtube_handle` are required and anything else is optional? What do you think about settling on what we really need to display a schedule and make all of that required as well?

Reply:
The "schedule" is always required, and either the twitch_handle or the youtube_handle (or both). 
The schedule has its own required fields that need to be  satisfied:

```
        "required": [
           "title",
           "start_date_utc",
           "duration_in_minutes",
           "platform",
           "stream_link"
         ]
```

Reply:
Oh, totally missed that. I rarely know the duration in advance, so I'm against requiring it. :)

Reply:
Removed duration from the required fields